maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,208 @@
+
+The existing https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin] has been designed to interact (mainly) with a running server instance. 
+We are here proposing to evolve the capabilities of this plugin for one to be able to build a fully configured server containing application deployment.
+ 
+This proposal adds a new kind of installation that is operated and controlled by the pom.xml file.  The pom.xml file contains all the information needed
+to control the server and application content and versioning.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* The Maven plugin produces inside the project target dir a directory (named ```server``` by default) that contains the server and any deployment.
+
+==== WildFly Maven Plugin new goals
+
+* ```wildfly:package```: Galleon provisioning of a server, CLI script execution, copy content to the server, deploy application (if present). 
+* ```wildfly:provision```: Galleon provisioning of a server.
+
+In addition to these 2 new goals, we are replacing the existing automatic download and unzip of wildfly-dist by a Galleon provisioning of 
+WildFly. This change impacts ```run``` and ```start```goals).
+ 
+==== Galleon related Plugin configuration items
+
+These configuration items are common to the ```provision``` and ```package``` goals. They are used to configure 
+Galleon to provision a server.
+
+* feature-packs: A list of feature-pack to install.
+* log-provisioning-time: Whether to log provisioning time at the end.
+* offline-provisioning: Whether to use offline mode when the plugin resolves an artifact.
+* galleon-options: Arbitrary Galleon options used when provisioning the server.
+* provisioning-dir: By default the server is generated in ```target/server``` directory. 
+The plugin offers a configuration item to provide the name of a directory relative to the project ```target``` directory in which the server will be installed.
+* provisioning-file: The path to the provisioning.xml file to use, by default ```<project base dir>/galleon/provisioning.xml``` is used. If some feature-packs have been set, 
+the provisioning file is being ignored.
+* record-state: Whether to record provisioned state in .galleon directory. When false, the file ```.wildfly-maven-plugin-provisioning.xml``` is generated in the server home directory.
+It contains Galleon provisioning informations used to provision this server.
+* layers: A set of layers to include when building-up a custom configuration.
+* excluded-layers: A set of layers to exclude when building-up a custom configuration.
+
+==== ```provision``` goal specific behavior
+
+Calling ```provision``` goal without any Galleon configuration items will provision the same default server 
+that we are provisioning when calling ```run``` and ```start``` goals.
+
+==== ```package``` goal specific configuration items and behavior
+
+A Galleon configuration is expected, if no galleon configuration is found, the goal execution fails.
```

Comment:
Just to be sure: This is about "Galleon related Plugin configuration item" and not only about providing value to `provisioning-file` option.

Reply:
@honza-kasik good point, I made it clear.